feeed/entropies.py

```diff
@@ -13,76 +18,224 @@ def __init__(self, feature_names='entropies'):
+        for trace in all_traces:
+            if trace not in previous_encountered:
+                previous_encountered.append(trace)
+                word = ""
+                for activity in trace:
+                    word += activity
+                    if word not in words:
+                        words.add(word)
+                        word = ""
+
+            N += len(trace)
+
+        N_w = len(words)
+        return round((N_w * math.log2(N)) / N,3)
 
     @classmethod
-    def entropy_lempel_ziv(cls, log_path):
-        return Entropies.default_call(log_path, "-z", "", "")
+    def entropy_lempel_ziv_flattened(cls,log):
+        unique_traces = list(variants_filter.get_variants(log))
+        N, N_w, words,previous_encountered = 0, 0, set(),[]
+        for trace in unique_traces:
+            if trace not in previous_encountered:
+                previous_encountered.append(trace)
+                word = ""
+                for activity in trace:
+                    word += activity
+                    if word not in words:
+                        words.add(word)
+                        word = ""
+
+            N += len(trace)
+
+        N_w = len(words)
+        return round((N_w * math.log2(N)) / N,3)
+        
 
     @classmethod
-    def entropy_k_block_diff_1(cls, log_path):
-        return Entropies.default_call(log_path, "-d", "1", "")
+    def entropy_k_block_diff_1(cls, log,k=1):
+        return round(Entropies.entropy_k_block(log, k) - Entropies.entropy_k_block(log, k-1),3)
 
     @classmethod
-    def entropy_k_block_diff_3(cls, log_path):
-        return Entropies.default_call(log_path, "-d", "3", "")
+    def entropy_k_block_diff_3(cls, log,k=3):
+        return round(Entropies.entropy_k_block(log, k) - Entropies.entropy_k_block(log, k-1),3)
 
     @classmethod
-    def entropy_k_block_diff_5(cls, log_path):
-        return Entropies.default_call(log_path, "-d", "5", "")
+    def entropy_k_block_diff_5(cls, log,k=5):
+        return round(Entropies.entropy_k_block(log, k) - Entropies.entropy_k_block(log, k-1),3)
 
     @classmethod
-    def entropy_k_block_ratio_1(cls, log_path):
-        return Entropies.default_call(log_path, "-r", "1", "")
+    def entropy_k_block_ratio_1(cls, log,k=1):
+        return round(Entropies.entropy_k_block(log, k)/k,3)
 
     @classmethod
-    def entropy_k_block_ratio_3(cls, log_path):
-        return Entropies.default_call(log_path, "-r", "3", "")
+    def entropy_k_block_ratio_3(cls, log,k=3):
+        return round(Entropies.entropy_k_block(log, k)/k,3)
 
     @classmethod
-    def entropy_k_block_ratio_5(cls, log_path):
-        return Entropies.default_call(log_path, "-r", "5", "")
+    def entropy_k_block_ratio_5(cls, log,k=5):
+        return round(Entropies.entropy_k_block(log, k)/k,3)
 
     @classmethod
-    def entropy_knn_3(cls, log_path):
-        return Entropies.default_call(log_path, "-k", "3", "1")
+    def entropy_knn_3(cls, log,k=3):
```

Comment:
I don't think we need the default parameter "k=3" here, since entropy_knn_3 will always be computed with k=3. Please remove that default parameter here `def entropy_knn_3(cls, log):` and specify in the call next line as: `Entropies.entropy_flattened_knn(log,k=3)` instead. Same for `entropy_knn_5` and `entropy_knn_7`.

Reply:
Also please add a comment in the following way:
`def entropy_knn_3(cls, log,k=3):#flattened`

Same for `entropy_knn_5` and `entropy_knn_7`.